presto-plan-checker-router-plugin/src/main/java/com/facebook/presto/router/scheduler/PlanCheckerRouterPluginPrestoClient.java

```
@@ -0,0 +1,243 @@
+            newErrorLocation = new ErrorLocation(
+                    errorLocation.getLineNumber(),
+                    errorLocation.getColumnNumber() - ANALYZE_CALL.length());
+        }
+        return new QueryError(
+                error.getMessage(),
+                error.getSqlState(),
+                error.getErrorCode(),
+                error.getErrorName(),
+                error.getErrorType(),
+                error.isRetriable(),
+                newErrorLocation,
+                new FailureInfo(
+                        failureInfo.getType(),
+                        failureInfo.getMessage(),
+                        failureInfo.getCause(),
+                        failureInfo.getSuppressed(),
+                        failureInfo.getStack(),
+                        newErrorLocation));
+    }
+
+    private ClientSession parseHeadersToClientSession(HttpServletRequest httpServletRequest)
+    {
+        SessionContext sessionContext = new HttpRequestSessionContext(
+                httpServletRequest,
+                new SqlParserOptions());
+
+        return new ClientSession(
+                planCheckerClusterURI,
+                sessionContext.getIdentity().getUser(),
+                sessionContext.getSource(),
+                sessionContext.getTraceToken(),
+                sessionContext.getClientTags(),
+                sessionContext.getClientInfo(),
+                sessionContext.getCatalog(),
+                sessionContext.getSchema(),
+                sessionContext.getTimeZoneId(),
+                Locale.ENGLISH,
+                getResourceEstimates(sessionContext),
+                sessionContext.getSystemProperties(),
+                sessionContext.getPreparedStatements(),
+                sessionContext.getIdentity().getRoles(),
+                sessionContext.getIdentity().getExtraCredentials(),
+                httpServletRequest.getHeader(PRESTO_TRANSACTION_ID),
+                new Duration(2, MINUTES),
```

Can we make the the client session timeout configurable ? Users may not want to wait 2 minutes; they may want to fail early